semgrep/semgrep/core_runner.py

```diff
@@ -578,8 +579,7 @@ def _run_rules_direct_to_semgrep_core(
             target_file.write(json.dumps(plan.to_json()))
             target_file.flush()
 
-            yaml = YAML()
-            yaml.dump({"rules": [rule._raw for rule in rules]}, rule_file)
+            rule_file.write(json.dumps({"rules": [rule._raw for rule in rules]}))
```

Comment:
This changes the pattern parse error locations/spans reported from core. But that might be okay, considering that with the YAML load-dump round trip we've probably already been changing the line numbers inadvertently. It's especially okay for registry rules where the user doesn't see the original rule file anyway.

So maybe we we pretty-format the JSON with indentation, and the patterns are still readable when we print it out in an error message, it's okay to change the source YAML to JSON here?